cpp/src/io/orc/orc_gpu.hpp

```diff
@@ -159,6 +159,7 @@ struct encoder_chunk_streams {
  * @brief Struct to describe a column stream within a stripe
  */
 struct StripeStream {
+  uint8_t* data_ptr;        // encoded and gathered output
```

Comment:
I think refactor to `std::byte` should be done more extensively in a separate PR, as `uint8_t` is used in many other places in ORC writer code.

Reply:
I agree with @ttnghia here, I'd like to do it in a separate PR so it can be done more systematically. The new implementation here is really just a replacement which has to interact with the rest of the writer (that consistently uses uint8_t).

Reply:
Works for me